src/funnels/uniV3/TickMath.sol

```diff
@@ -0,0 +1,258 @@
+// SPDX-License-Identifier: GPL-3.0
+
+// Based on https://github.com/gelatodigital/g-uni-v1-core/blob/bea63422e2155242b051896b635508b7a99d2a1a/contracts/vendor/uniswap/TickMath.sol
+// Uniswap version - https://github.com/Uniswap/v3-core/blob/412d9b236a1e75a98568d49b1aeb21e3a1430544/contracts/libraries/TickMath.sol
```

Comment:
All the math in this file was unchecked in the original Uniswap version--assuming that was safe, you should be able to wrap the two function bodies below in `unchecked` blocks and save a decent amount of gas.